### PR TITLE
code review

### PR DESCRIPTION
**Pull request by lamtung16**
@tdhock, could you please review the code section I made some new comments

### CODE REVIEW

File: MLP.py [hunk 1]
```diff
@@ -48,19 +48,19 @@ def cv_learn(n_splits, X, y, n_hiddens, layer_size, batch_size, n_ite):
     loss_func = SquaredHingeLoss()
 
     # learn best ite
-    total_losses = {'train': np.zeros(n_ite), 'val': np.zeros(n_ite)}
-    data_splits = {'X_train': [], 'X_val': [], 'y_train': [], 'y_val': []}
+    total_losses = {'subtrain': np.zeros(n_ite), 'val': np.zeros(n_ite)}
+    data_splits = {'X_subtrain': [], 'X_val': [], 'y_subtrain': [], 'y_val': []}
```
Comment (tdhock) [inline MLP.py:51]: please remove this line, and instead initialize to empty dict then populate later with a for loop over sets

File: MLP.py [hunk 2]
```diff
@@ -48,19 +48,19 @@ def cv_learn(n_splits, X, y, n_hiddens, layer_size, batch_size, n_ite):
...
-        data_splits['y_train'].append(y[subtrain_idx])
-        data_splits['y_val'].append(y[val_idx])
+        indices = {'subtrain': subtrain_idx, 'val': val_idx}
+        for key in data_splits.keys():
+            feature_target, set_type = key.split('_')       # (X or y) and (subtrain or val)
+            data_splits[key].append(X[indices[set_type]] if feature_target == 'X' else y[indices[set_type]])
 
         # Create DataLoader
-        dataset    = TensorDataset(data_splits['X_train'], data_splits['y_train'])
+        dataset    = TensorDataset(data_splits['X_subtrain'][-1], data_splits['y_subtrain'][-1])
```
Comment (tdhock) [inline MLP.py:62]: why -1?

File: MLP.py [hunk 3]
```diff
@@ -72,21 +72,21 @@ def cv_learn(n_splits, X, y, n_hiddens, layer_size, batch_size, n_ite):
...
             # training
-            train_loss = 0
+            subtrain_loss = 0
             for inputs, labels in dataloader:
                 optimizer.zero_grad()
                 loss = loss_func(model(inputs), labels)
                 loss.backward()
                 optimizer.step()
-                train_loss += loss.item()
+                subtrain_loss += loss.item()
```
Comment (tdhock) [inline MLP.py]: instead of computing loss of each batch and then summing, please compute the loss of both subtrain/validation sets, at the end of every epoch, like we did in class (less noisy than summing loss over batches)

File: MLP.py (hunk 1 above)
Comment (lamtung16) [reply MLP.py:51]: Ok will do.

File: MLP.py (hunk 2 above)
Comment (lamtung16) [reply MLP.py:62]: Because of the shape (simply, I need [X], not [[X]]). That number can be 0 (instead of -1).

File: MLP.py (hunk 3 above)
Comment (lamtung16) [reply MLP.py]: Ok will do.
